moped-editor/src/views/projects/projectView/ProjectComponents/utils/features.js

```diff
@@ -100,22 +100,48 @@ export const getDrawId = (feature) => feature?.properties?.DRAW_ID;
 
 /**
  * Determine if a draft feature is drawn by the user by the draw ID
- * that is added when creating a new drawn feature
+ * that is added when creating a drawn feature that is not yet saved
  * @param {Object} feature - a feature object
  * @returns {Boolean} - true if feature is drawn by user
  */
 export const isDrawnDraftFeature = (feature) =>
   getDrawId(feature) ? true : false;
 
 /**
- * Determine if a feature is drawn by the user by checking that the source is
- * not CTN
+ * Determine if any features captured in a Mapbox click event are features
+ * that are a drawn and not yet saved
+ * @param {Array} features - array for features from a Mapbox click event
+ * @returns {Boolean} - true if any feature is drawn by user and already saved
+ */
+export const areAnyClickedFeaturesDraftDrawnFeatures = (clickedFeatures) =>
+  clickedFeatures?.some((feature) => isDrawnDraftFeature(feature));
+
+/**
+ * Add fetched component feature's source layer ("drawnByUserLine" or "drawnByUserPoint")
+ * to feature properties so we can differentiate them from non-drawn features
+ * @param {Object} feature - a feature object
+ */
+export const addDrawnFeatureProperties = (feature) => {
+  feature.properties.sourceLayer = feature.source_layer;
+};
+
+/**
+ * Determine if a feature was saved previously by checking for the source layer
+ * data that is only present in saved drawn features
  * @param {Object} feature - a feature object
  * @returns {Boolean} - true if feature is drawn by user
  */
-export const isDrawnExistingFeature = (feature) =>
-  !feature.source.includes("ATD_ADMIN.CTN_Intersections") &&
-  !feature.source.includes("ATD_ADMIN.CTN");
+export const isDrawnSavedFeature = (feature) =>
+  feature.properties.sourceLayer?.includes("drawnByUser");
```

Comment:
not at all in scope of this PR, but i kind of want to make a check constraint that enforces the `drawnByUser` value in the feature tables in the DB. it's weird that we can put any string we want to and it would cause big problems.

Reply:
yeah, this feels like it needs to tightening up to me too! Took me a bit to remember that this string is set in the frontend code (`makeDrawnFeature`), and it is hard to understand how this fits into the big picture.